File: tests/workflow/consumer/rust/src/consumer/broadcasts/structb.rs
use super::{protocol, Consumer, Context};
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::StructB,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.broadcast.structb += 1;
    context.broadcast.check();
}

File: experimental/workflow_mt/producer/typescript/src/producer/implementation/beacons/structb.ts
import { Producer, Identification, Filter, Context, Protocol } from "./index";
import { emit } from "../../beacons/structb";

export function handler(
    beacon: Protocol.StructB,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer,
    sequence: number
): Promise<void> {
    return emit(beacon, consumer, filter, context, producer).then(() => {
        const confirmation =
            new Protocol.InternalServiceGroup.BeaconConfirmation({
                error: undefined,
            });
        return producer.send(
            consumer.uuid(),
            confirmation.pack(sequence, consumer.uuid())
        );
    });
}